CMake build files with opus linking, works in bam

```
[ 98%] Linking CXX executable DDNet
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_get_next_page':
/root/opusfile-0.6/src/opusfile.c:191: undefined reference to `ogg_sync_pageseek'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_get_data':
/root/opusfile-0.6/src/opusfile.c:149: undefined reference to `ogg_sync_buffer'
/root/opusfile-0.6/src/opusfile.c:152: undefined reference to `ogg_sync_wrote'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_collect_audio_packets':
/root/opusfile-0.6/src/opusfile.c:776: undefined reference to `ogg_stream_packetout'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_find_initial_pcm_offset':
/root/opusfile-0.6/src/opusfile.c:862: undefined reference to `ogg_page_bos'
/root/opusfile-0.6/src/opusfile.c:873: undefined reference to `ogg_page_serialno'
/root/opusfile-0.6/src/opusfile.c:874: undefined reference to `ogg_stream_pagein'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_fetch_headers_impl':
/root/opusfile-0.6/src/opusfile.c:495: undefined reference to `ogg_page_serialno'
/root/opusfile-0.6/src/opusfile.c:495: undefined reference to `ogg_stream_reset_serialno'
/root/opusfile-0.6/src/opusfile.c:496: undefined reference to `ogg_stream_pagein'
/root/opusfile-0.6/src/opusfile.c:497: undefined reference to `ogg_stream_packetout'
/root/opusfile-0.6/src/opusfile.c:482: undefined reference to `ogg_page_bos'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_lookup_page_serialno':
/root/opusfile-0.6/src/opusfile.c:264: undefined reference to `ogg_page_serialno'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_add_serialno':
/root/opusfile-0.6/src/opusfile.c:235: undefined reference to `ogg_page_serialno'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_fetch_headers_impl':
/root/opusfile-0.6/src/opusfile.c:517: undefined reference to `ogg_page_serialno'
/root/opusfile-0.6/src/opusfile.c:520: undefined reference to `ogg_stream_packetout'
/root/opusfile-0.6/src/opusfile.c:552: undefined reference to `ogg_stream_packetout'
/root/opusfile-0.6/src/opusfile.c:536: undefined reference to `ogg_page_bos'
/root/opusfile-0.6/src/opusfile.c:531: undefined reference to `ogg_page_serialno'
/root/opusfile-0.6/src/opusfile.c:532: undefined reference to `ogg_stream_pagein'
/root/opusfile-0.6/src/opusfile.c:517: undefined reference to `ogg_stream_pagein'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_seek_helper':
/root/opusfile-0.6/src/opusfile.c:164: undefined reference to `ogg_sync_reset'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_get_last_page':
/root/opusfile-0.6/src/opusfile.c:436: undefined reference to `ogg_page_serialno'
/root/opusfile-0.6/src/opusfile.c:440: undefined reference to `ogg_page_granulepos'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_clear':
/root/opusfile-0.6/src/opusfile.c:1479: undefined reference to `ogg_stream_clear'
/root/opusfile-0.6/src/opusfile.c:1480: undefined reference to `ogg_sync_clear'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_open1':
/root/opusfile-0.6/src/opusfile.c:1498: undefined reference to `ogg_sync_init'
/root/opusfile-0.6/src/opusfile.c:1529: undefined reference to `ogg_stream_init'
/root/opusfile-0.6/src/opusfile.c:1508: undefined reference to `ogg_sync_buffer'
/root/opusfile-0.6/src/opusfile.c:1510: undefined reference to `ogg_sync_wrote'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_test':
/root/opusfile-0.6/src/opusfile.c:89: undefined reference to `ogg_sync_init'
/root/opusfile-0.6/src/opusfile.c:90: undefined reference to `ogg_sync_buffer'
/root/opusfile-0.6/src/opusfile.c:96: undefined reference to `ogg_sync_wrote'
/root/opusfile-0.6/src/opusfile.c:97: undefined reference to `ogg_stream_init'
/root/opusfile-0.6/src/opusfile.c:101: undefined reference to `ogg_sync_pageout'
/root/opusfile-0.6/src/opusfile.c:106: undefined reference to `ogg_page_serialno'
/root/opusfile-0.6/src/opusfile.c:106: undefined reference to `ogg_stream_reset_serialno'
/root/opusfile-0.6/src/opusfile.c:107: undefined reference to `ogg_stream_pagein'
/root/opusfile-0.6/src/opusfile.c:110: undefined reference to `ogg_stream_packetout'
/root/opusfile-0.6/src/opusfile.c:124: undefined reference to `ogg_stream_clear'
/root/opusfile-0.6/src/opusfile.c:127: undefined reference to `ogg_sync_clear'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_fetch_and_process_page':
/root/opusfile-0.6/src/opusfile.c:1850: undefined reference to `ogg_page_serialno'
/root/opusfile-0.6/src/opusfile.c:1853: undefined reference to `ogg_page_bos'
/root/opusfile-0.6/src/opusfile.c:1934: undefined reference to `ogg_stream_pagein'
/root/opusfile-0.6/src/opusfile.c:1889: undefined reference to `ogg_stream_reset_serialno'
/root/opusfile-0.6/src/opusfile.c:1878: undefined reference to `ogg_page_serialno'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_pcm_seek_page':
/root/opusfile-0.6/src/opusfile.c:2328: undefined reference to `ogg_page_serialno'
/root/opusfile-0.6/src/opusfile.c:2390: undefined reference to `ogg_stream_reset_serialno'
/root/opusfile-0.6/src/opusfile.c:2329: undefined reference to `ogg_page_granulepos'
/root/opusfile-0.6/src/opusfile.c:2390: undefined reference to `ogg_stream_reset_serialno'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_open_seekable2':
/root/opusfile-0.6/src/opusfile.c:1427: undefined reference to `ogg_sync_init'
/root/opusfile-0.6/src/opusfile.c:1428: undefined reference to `ogg_stream_init'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_get_prev_page_serial':
/root/opusfile-0.6/src/opusfile.c:345: undefined reference to `ogg_page_serialno'
/root/opusfile-0.6/src/opusfile.c:355: undefined reference to `ogg_page_granulepos'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_open_seekable2':
/root/opusfile-0.6/src/opusfile.c:1431: undefined reference to `ogg_stream_clear'
/root/opusfile-0.6/src/opusfile.c:1432: undefined reference to `ogg_sync_clear'
../other/opus/linux/lib64/libopusfile.a(opusfile.o): In function `op_bisect_forward_serialno':
/root/opusfile-0.6/src/opusfile.c:1201: undefined reference to `ogg_page_serialno'
/root/opusfile-0.6/src/opusfile.c:1202: undefined reference to `ogg_page_granulepos'
collect2: error: ld returned 1 exit status
CMakeFiles/DDNet.dir/build.make:1773: recipe for target 'DDNet' failed
make[2]: *** [DDNet] Error 1
CMakeFiles/Makefile2:72: recipe for target 'CMakeFiles/DDNet.dir/all' failed
make[1]: *** [CMakeFiles/DDNet.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```

config.lua:
```
compiler.driver = 'gcc'
compiler.c_compiler = false
compiler.cxx_compiler = false
stackprotector.value = true
minmacosxsdk.value = false
macosxppc.value = false
zlib.value = true
zlib.include_path = false
sdl.value = true
sdl.use_pkgconfig = true
sdl.use_winlib = 0
sdl.use_osxframework = false
freetype.value = true
freetype.use_pkgconfig = true
freetype.use_winlib = 0
curl.value = true
curl.use_pkgconfig = true
opusfile.value = true
opusfile.use_pkgconfig = false
opus.value = true
opus.use_pkgconfig = false
ogg.value = true
ogg.use_pkgconfig = false
mysql.value = true
mysql.use_mysqlconfig = false
websockets.value = false
```

``
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.28") 
-- Checking for module 'libcurl'
--   Found libcurl, version 7.37.0
-- Found Curl: /usr/lib64/libcurl.so  
-- Checking for module 'freetype2'
--   Found freetype2, version 18.3.12
-- Found Freetype: /usr/lib64/libfreetype.so  
-- Checking for module 'ogg'
--   Package 'ogg' not found
-- Found Ogg: /home/deen/git/ddnet/other/opus/include/ogg  
-- Checking for module 'opus'
--   Package 'opus' not found
-- Found Opus: /home/deen/git/ddnet/other/opus/include/opus  
-- Checking for module 'opusfile'
--   Package 'opusfile' not found
-- Found Opusfile: /home/deen/git/ddnet/other/opus/linux/lib64/libopusfile.a  
-- Found PythonInterp: /usr/bin/python (found version "2.7.9") 
-- Checking for module 'sdl2'
--   Found sdl2, version 2.0.4
-- Found SDL2: /usr/local/lib/libSDL2.so  
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.8") 
-- ******** DDNet ********
-- Build type: Release
-- Compiler: /usr/bin/c++
-- Dependencies:
--  * Curl found
--  * Freetype found
--  * Ogg not found (using bundled version)
--  * Opus not found (using bundled version)
--  * Opusfile not found (using bundled version)
--  * PythonInterp found
--  * SDL2 found
--  * Zlib found
-- Performing Test DEFINE_FORTIFY_SOURCE
-- Performing Test DEFINE_FORTIFY_SOURCE - Success
-- Configuring done
CMake Error at CMakeLists.txt:378 (target_sources):
  Error evaluating generator expression:

    $<TARGET_OBJECTS:png>

  The evaluation of the TARGET_OBJECTS generator expression is only suitable
  for consumption by CMake.  It is not suitable for writing out elsewhere.


CMake Error at CMakeLists.txt:378 (target_sources):
  Error evaluating generator expression:

    $<TARGET_OBJECTS:png>

  The evaluation of the TARGET_OBJECTS generator expression is only suitable
  for consumption by CMake.  It is not suitable for writing out elsewhere.


CMake Error at CMakeLists.txt:378 (target_sources):
  Error evaluating generator expression:

    $<TARGET_OBJECTS:png>

  The evaluation of the TARGET_OBJECTS generator expression is only suitable
  for consumption by CMake.  It is not suitable for writing out elsewhere.


CMake Error at CMakeLists.txt:378 (target_sources):
  Error evaluating generator expression:

    $<TARGET_OBJECTS:png>

  The evaluation of the TARGET_OBJECTS generator expression is only suitable
  for consumption by CMake.  It is not suitable for writing out elsewhere.


CMake Error at CMakeLists.txt:378 (target_sources):
  Error evaluating generator expression:

    $<TARGET_OBJECTS:png>

  The evaluation of the TARGET_OBJECTS generator expression is only suitable
  for consumption by CMake.  It is not suitable for writing out elsewhere.


-- Generating done
-- Build files have been written to: /home/deen/git/ddnet/build
``
[CMakeCache.txt](https://github.com/ddnet/ddnet/files/850148/CMakeCache.txt)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -120,7 +120,11 @@ if(TARGET_OS STREQUAL "windows")
   set(PLATFORM_LIBS ws2_32) # Windows sockets
 else()
   set(PLATFORM_CLIENT_LIBS GL GLU X11)
-  set(PLATFORM_LIBS)
+  if(TARGET_OS STREQUAL "linux")
+    set(PLATFORM_LIBS rt) # clock_gettime for glibc < 2.17
+  else()
+    set(PLATFORM_LIBS)
+  endif()
 endif()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES Clang OR CMAKE_CXX_COMPILER_ID MATCHES GNU)
@@ -276,10 +280,13 @@ if(CLIENT)
     ${LIBS}
     ${CURL_LIBRARIES}
     ${FREETYPE_LIBRARIES}
-    ${OGG_LIBRARIES}
+    ${SDL2_LIBRARIES}
+
+    # Order of these three is important.
     ${OPUSFILE_LIBRARIES}
     ${OPUS_LIBRARIES}
-    ${SDL2_LIBRARIES}
+    ${OGG_LIBRARIES}
+
     ${PLATFORM_CLIENT_LIBS}
   )
 

